packages/ui/src/ui/pages/query-tracker/module/query_result/actions.ts

```diff
@@ -242,3 +257,55 @@ export function updateQueryResult(
         }
     };
 }
+
+export function loadQueryResultsErrors(
+    query: QueryItem,
+): ThunkAction<any, RootState, any, QueryResultsActions> {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {result_count} = query;
+
+        if (!result_count) return;
+
+        const inds: number[] = [];
+        for (let ind = 0; ind < result_count; ind++) {
+            if (!hasQueryResult(state, query.id, ind)) inds.push(ind);
+        }
+
+        if (inds.length === 0) return;
+
+        try {
+            const {results, error, errorIndices} = await wrapApiPromiseByToaster(
+                dispatch(getQueryResultMetaList(query.id, inds)),
+                {
+                    toasterName: `load_query_results`,
+                    skipSuccessToast: true,
+                    errorTitle: `Failed to load query results meta`,
+                },
+            );
+            const resultsErrors: {[index: number]: QueryResultErrorState} = {};
+            forEach_(results, (meta) => {
+                if (meta.error) {
+                    resultsErrors[meta.result_index] = {
+                        error: meta.error as Error,
+                        state: QueryResultState.Error,
+                        resultReady: false,
+                    };
+                }
+            });
+            if (error) {
+                forEach_(error.inner_errors, (e, ind) => {
+                    resultsErrors[inds[errorIndices[ind]]] = {
```

Comment:
`resultsErrors[inds[errorIndices[ind]]]` 
It is hard to read code-sentences like this, also it looks like a probable source of unwanted exceptions. The code would be more clear and compact if `getQueryResultMetaList` returned batch result "as is" without handling by `splitBatchResult`.